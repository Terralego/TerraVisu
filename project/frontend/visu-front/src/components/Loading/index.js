import React from 'react';
import { Spinner } from '@blueprintjs/core';

export const Loading = props => <Spinner intent="primary" {...props} />;

export default Loading;
