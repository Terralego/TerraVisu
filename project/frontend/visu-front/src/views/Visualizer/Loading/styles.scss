@import '../../variables.scss';

.visuPlaceholder {
  position: absolute;
  top: 0;
  left: 3.35rem;
  width: calc(100% -3rem);
  height: 100%;
  display: flex;
  flex-direction: row;
  z-index: 80;

  &__layerstree {
    width: 19.5rem;
    height: 100%;
    background-color:#30404d;
    box-shadow: 9px 8px 24px 0 rgba(16,22,26,0.40);
    padding: .3rem 0 .5rem .8rem;

    &-appName, &-title {
      width: calc(100% - 8rem);
      height: 1.8rem;
      margin-bottom: 1.5rem;
      padding: .3rem 0 .5rem .8rem;
    }

    &-appName {
      width: calc(100% - 2rem);
      height: 2.2rem;
      margin-bottom: .5rem;
    }

    &-group {
      margin-top: 1.5rem;
    }

    &-group-title {
      width: calc(100% - 6rem);
      height: 1.4rem;
    }
    &-group-items {
      width: calc(100% - 1.5rem);
      height: auto;
      padding: 0 1.5rem 0 1.2rem;
    }
    &-group-item {
      list-style: none;
      width: 100%;
      height: 2rem;
      border: 1px solid #30404D;
    }
  }
}
.skeleton {
  background: linear-gradient(to left, #30404d, #394b59);
  background-size: 200%;
  animation: 1s linear 0s alternate infinite placeholderShimmer;
}

@keyframes placeholderShimmer {
  from { background-position: 0 0; }
  to { background-position: 100% 0; }
}

@media(min-width: $breakpoints-tablet) and (max-width: $breakpoints-desktop) and (orientation: landscape) {
  .visuPlaceholder {
    left: 2.95rem;
    &__layerstree {
      width: 17.5rem;
    }
  }
}
@media(min-width: $breakpoints-phone) and (max-width: $breakpoints-tablet) and (orientation: portrait) {
  .visuPlaceholder {
    left: 2.85rem;
    &__layerstree {
      width: 14rem;
    }
  }
}
