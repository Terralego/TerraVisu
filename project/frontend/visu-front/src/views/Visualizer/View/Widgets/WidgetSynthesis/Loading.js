import React from 'react';

const Loading = () => (
  <div className="widget-synthesis__loading">â€¦</div>
);
export default Loading;
