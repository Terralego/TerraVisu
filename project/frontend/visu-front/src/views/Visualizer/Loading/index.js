import React from 'react';
import './styles.scss';

export const Loading = () => (
  <div className="visuPlaceholder">
    <div className="visuPlaceholder__layerstree">
      <div className="visuPlaceholder__layerstree-appName skeleton" />
      <div className="visuPlaceholder__layerstree-title skeleton" />
      <div className="visuPlaceholder__layerstree-group">
        <div className="visuPlaceholder__layerstree-group-title skeleton" />
        <ul className="visuPlaceholder__layerstree-group-items">
          <li className="visuPlaceholder__layerstree-group-item skeleton" />
          <li className="visuPlaceholder__layerstree-group-item skeleton" />
          <li className="visuPlaceholder__layerstree-group-item skeleton" />
        </ul>
      </div>
      <div className="visuPlaceholder__layerstree-group">
        <div className="visuPlaceholder__layerstree-group-title skeleton" />
        <ul className="visuPlaceholder__layerstree-group-items">
          <li className="visuPlaceholder__layerstree-group-item skeleton" />
          <li className="visuPlaceholder__layerstree-group-item skeleton" />
        </ul>
      </div>
      <div className="visuPlaceholder__layerstree-group">
        <div className="visuPlaceholder__layerstree-group-title skeleton" />
        <ul className="visuPlaceholder__layerstree-group-items">
          <li className="visuPlaceholder__layerstree-group-item skeleton" />
        </ul>
      </div>
    </div>
  </div>
);

export default Loading;
