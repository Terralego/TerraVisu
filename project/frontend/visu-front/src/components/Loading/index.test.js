import React from 'react';
import renderer from 'react-test-renderer';
import Loading from '.';

it('should render correctly', () => {
  const tree = renderer.create(<Loading />).toJSON();
  expect(tree).toMatchSnapshot();
});
