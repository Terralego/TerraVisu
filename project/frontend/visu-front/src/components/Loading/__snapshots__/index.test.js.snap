// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="bp3-spinner bp3-intent-primary"
>
  <div
    className="bp3-spinner-animation"
  >
    <svg
      height={50}
      strokeWidth="8.00"
      viewBox="1.00 1.00 98.00 98.00"
      width={50}
    >
      <path
        className="bp3-spinner-track"
        d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
      />
      <path
        className="bp3-spinner-head"
        d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
        pathLength={280}
        strokeDasharray="280 280"
        strokeDashoffset={210}
      />
    </svg>
  </div>
</div>
`;
